### ISSUE TITLE
Unable to change from a premium style back to default during signup

### ISSUE THREAD

**Issue by jornp** [Automattic/wp-calypso#95578]
### Quick summary

When choosing the Free plan during onboarding and selecting a premium style, the Launchpad correctly states that an upgrade is required:

![Image](https://github.com/user-attachments/assets/ae389a08-ca49-472f-bb10-9c9354ce69c1)

However, after that, I'm unable to switch back to the default/free style. The “remove your premium styles” link in the screenshot above links to [this support document](https://wordpress.com/support/using-styles/#reset-all-styles), containing instructions for the editor that can't easily be applied from the Launchpad.

When I go back to the "Select a design" step, choose the theme again, and select the default style, the selection is ignored and the premium style is still applied. (Switching to a different premium style works, though.)

### Steps to reproduce

https://github.com/user-attachments/assets/f42c783b-2a80-4eaf-966c-521811539a2b

1. Go to /start and continue to the theme selection step.
2. Choose a theme and select a premium style.
3. Click Continue and then click "Decide later" in the upsell dialog.
4. In the Launchpad, click "Select a design" to go back.
5. Choose the same theme, select the default style, and click Continue.
6. The previously-chosen premium styles are still applied.

### What you expected to happen

1. I expected the “remove your premium styles” link to revert the theme back to its default style.
2. I expected to be able to switch back to the default style by going back to "Select a design".

### What actually happened

1. The “remove your premium styles” link points to a support document with in-editor instructions which can't be applied easily from the Launchpad.
2. After going back to "Select a design" and selecting the default style, the premium style is still applied.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Logs or notes

This was discovered during a self-serve interview (pxLjZ-93c-p2).

I'd love to help, and I'd be happy to write the fix myself, but I'd love some pointers about where/how to fix it exactly.

I'm not exactly sure how the default style is treated. I took a quick look, and I think the theme and its style are applied in [setDesignOnSite](https://github.com/Automattic/wp-calypso/blob/7ca5a84c043b419cfc6753d425da013bdcc6af98/packages/data-stores/src/site/actions.ts#L376). However, it looks like `getGlobalStylesVariations` only gets the _variations_ and not the default style. Does the default case need to be handled there, or is reverting to default a matter of _removing_ the chosen style variation? Thanks!

**Comment by gcsecsey** [comment]
I could replicate this issue and look into it briefly. One possible workaround is clicking the "Edit site design" link on Launchpad, then in Gutenberg, going to Styles > Reset Styles, then clicking the WordPress icon in the top left to return to Launchpad. This clears the "Choose a plan" task on the Launchpad sidebar. However, as you also mentioned, this is very unintuitive.

**Comment by gcsecsey** [comment]
The [setDesignOnSite](https://github.com/Automattic/wp-calypso/blob/7ca5a84c043b419cfc6753d425da013bdcc6af98/packages/data-stores/src/site/actions.ts#L376) action is only dispatched when the "Continue" button is clicked on a style page.

I could reproduce this issue a bit earlier within the unified design picker, by just visiting a premium style page, going back, then applying a free style. I think throwing the modal in this case is also not expected:

https://github.com/user-attachments/assets/fa461fff-4e18-4a78-bad1-c8e1c69faae1

**Comment by jornp** [comment]
Thank you for looking into this, @gcsecsey! Thanks for writing out the workaround. While it's something a new user likely wouldn't find, it is a valid solution!

Just checking — in your video, isn't the Ember style you're selecting a premium style too, making the modal correct?

**Comment by gcsecsey** [comment]
> Just checking — in your video, isn't the Ember style you're selecting a premium style too, making the modal correct?

Yes, it is. I'm sorry but missed the premium indicator under the TwentyTwentyFour styles section, so yes the modal is correct.

**Comment by jornp** [comment]
No worries at all!

I see you self-assigned the issue — let me know if you need any help! I'd be happy to discuss or help with the implementation or testing. Feel free to ping me here or on Slack. 🙂

**Comment by gcsecsey** [comment]
It was a bit hard to try to reproduce the issue locally - for some reason a couple of times, the selected theme is hidden in the unified design picker for me when going back to select a new style:

https://github.com/user-attachments/assets/84d66648-5d56-4757-9f13-b6cc13125451

I raised a PR which I _think_ resolved the original issue, but I'll need to test it on the Calypso environment to make sure.

**Comment by jornp** [comment]
Thanks for the fix attempt! I tested the PR (it's almost perfect!) and left a comment.

In your video, I think you're unable to find your selected theme because I think it's under the Business tab. When you go back to "Select a design" around 0:30, you're brought back to the Blog tab. Perhaps you can find Attar under the Business tab again?

**Comment by p-jackson** [comment]
> ### What you expected to happen
> 1. I expected the “remove your premium styles” link to revert the theme back to its default style.

@jornp w.r.t. this expectation from the original issue description, @dsas provided some useful context about how not providing a one-click fix outside the editor is "by design" pekYwv-1qM-p2#comment-1021. Adding a one-click fix isn't a "bug fix" in that sense and would be a new feature request.

The other issues raised here (like going back to the design picker and it being difficult to get back to the launchpad) are bugs though.

**Comment by p-jackson** [comment]
I've deployed https://github.com/Automattic/wp-calypso/pull/95618 which fixes the most basic issue where setting the default style in the picker wasn't setting the default style on the actual site (and thus leaving the warning in the launchpad).

Still to fix is
- Choosing a theme like "Impressionist" (which doesn't have any variations) doesn't clear the warning in launchpad
- The variation selection bug: If the default variation is already selected, you have to explicitly click the default variation again before it will correctly clear the existing style variation

**Comment by candy02058912** [comment]
I've tested with the scenario

> Choosing a theme like "Impressionist" (which doesn't have any variations) doesn't clear the warning in launchpad

This part seems fixed as I can't reproduce it

https://github.com/user-attachments/assets/cc3a97e2-a97f-45b1-8ced-d5854a868b23
